ckan/config/solr/schema.xml

```diff
@@ -24,29 +24,29 @@
 <!-- We update the version when there is a backward-incompatible change to this
 schema. In this case the version should be set to the next CKAN version number.
 (x.y but not x.y.z since it needs to be a float) -->
-<schema name="ckan" version="2.9">
+<schema name="ckan" version="2.10">
```

Comment:
I get this when trying to use this schema:
```
  File "/home/adria/dev/pyenvs/ckan-py3/src/ckan/ckan/config/environment.py", line 176, in update_config
    search.check_solr_schema_version()
  File "/home/adria/dev/pyenvs/ckan-py3/src/ckan/ckan/lib/search/__init__.py", line 318, in check_solr_schema_version
    version = tree.documentElement.getAttribute('version')
ckan.lib.search.common.SearchError: SOLR schema version not supported: 2.1. Supported versions are [2.8, 2.9, 2.10]
```
Did you get that working at all? I'm curious to know how this could work on your end and the tests.

The reason is failing is because Solr must be parsing this value like a float and when you request the schema using its endpoint it returns `<schema name="ckan" version="2.1">`

I know what you are thinking, let's just add a `v` and be done with it:
```
<schema name="ckan" version="v2.10">
```
That doesn't work and makes Solr crash. In fact, it seems like [we shouldn't have used this field at all](https://cwiki.apache.org/confluence/display/solr/schemaxml#SchemaXml-Schemaversionattributeintherootnode). Adding custom fields or attributes to the schema doesn't work because solr strips everything it doesn't know of away when requesting the schema.

I guess the only safe option is to use the `name` attribute, which on the Solr side is just used for displaying purposes, with a convention like `ckan-{version_number}`, eg `<schema name="ckan-2.10" version="1.6">`. And modify our schema version check here:

https://github.com/ckan/ckan/blob/157aa9b5c78da005f0938eb2037b67bd774f5cce/ckan/lib/search/__init__.py#L317

To use the `name` attribute. We can fallback to the `version` attribute if using a 2.8/2.9 schema.

Reply:
The comment just above that line points to the issue:  `(x.y but not x.y.z since it needs to be a float) -->`

Let's just go with 3.0? Aren't we on 3.0 at this point?

Reply:
3.0 would have major breaking changes, current master does not have those compared to 2.9 so most likely next release will be 2.10

Reply:
Wasn't python2 support removed?

Reply:
Most of the breaking changes are already in 2.9, but anyway there isn't an official decision what the next version is called, it is 2.10 in the code base for now.

Reply:
There's a PR in master removing python 2 support (#5172), which was not backported to 2.9. If that's not a major breaking change, I don't know what is.

Reply:
But still, the code base says that the next version is 2.10. https://github.com/ckan/ckan/blob/f96337c172c4111e6228f8e12bdea82a80139814/ckan/__init__.py#L3 This PR is not really the place to decide otherwise :p

Reply:
Ok, then I guess we don't need to update the schema version, if it's compatible with unmodified ckan.

Reply:
@smotornyuk regardless of what we end up using here, I still want to understand why CKAN is running at all in this PR and not failing with `SOLR schema version not supported: 2.1. Supported versions are [2.8, 2.9, 2.10]` :)

Reply:
Hm... I think that I've built docker image **before** I've changed the version inside `schema.xml`, thus it still uses `2.9` which is parsed with no problems.

Reply:
Yes, that's probably what happened, the Cypress tests failed because they used the updated version. 
I've pushed some changes (9fcac69) to switch to storing the version in the `name` attribute. I think even if CKAN would work without changing the version we should do it as the syntax is Solr 8 only. We can backport it to 2.9.5 to allow 2.9 to run on this version of the schema.

We can have the 2.10 vs 3.0 discussion elsewhere, it should be easy to change the numbers when we settle on a version